packages/server/api.js

```diff
@@ -891,8 +833,78 @@ module.exports = async function attachAPI(app, {wss, db, dbDir}) {
     ])
   ])
 
+  app.post('/api/users', [
+    ...middleware.loadVarFromBody('username'),
+    ...middleware.requireNameValid('username'),
+    ...middleware.loadVarFromBody('password'),
+    ...middleware.validateVar('password', validate.string),
+
+    async (request, response) => {
+      const { username, password } = request[middleware.vars]
+
+      if (await db.users.findOne({username})) {
+        response.status(500).json({
+          error: errors.NAME_ALREADY_TAKEN
+        })
+
+        return
+      }
+
+      if (password.length < 6) {
+        response.status(400).json({
+          error: errors.SHORT_PASSWORD
```

Comment:
`{ expectedLength: 6 }`?

Reply:
Nah, that's already in the error message. We can make minimum required password length a setting in the future, but since it's hard-coded as 6 characters minimum, there's no need to add an extra `expectedLength` property.